src/renderer/html_handlebars/hbs_renderer.rs

```diff
@@ -324,16 +357,16 @@ fn build_header_links(html: String, filename: &str) -> String {
             }
             let id = id.chars()
                 .filter_map(|c| if c.is_alphanumeric() || c == '-' || c == '_' {
-                                if c.is_ascii() {
-                                    Some(c.to_ascii_lowercase())
-                                } else {
-                                    Some(c)
-                                }
-                            } else if c.is_whitespace() && c.is_ascii() {
-                                Some('-')
-                            } else {
-                                None
-                            })
+                    if c.is_ascii() {
```

Comment:
Yeah, definitely. I was mainly trying to break things up into smaller chunks so it's easier to understand what `HtmlHandlebars::render()` does, but I may as well clean this up too.